### PR TITLE
Update Python base version to 3.7+

### PR DESCRIPTION
**Pull request by npradeep357**
Updated Python version to 3.7+ as 3.6 came to end-of-life support. Updated tests
related to #128

### CODE REVIEW

**Comment by npradeep357** [comment]
@ccordoba12, updated Python version to 3.7+ and removed python 3.6 slots on Github actions.
Please approve workflows

**Comment by npradeep357** [comment]
@ccordoba12 or @krassowski or authors, I see issue in static code analysis checks but I think that's not related to this PR. It says validate JSON schema check failed. I just updated the static.yml to use python 3.7 but not sure what's happening?
Can someone point out to me what's the issue here.

image

Thank you

**Comment by npradeep357** [comment]
@krassowski if you are available, can you please approve the workflows for checks.

Thank you

**Comment by npradeep357** [comment]
Only difference is that the static checks now use py 3.7 instead of 3.6. 
Will jsonschema won't work as is in 3.7? or Is there a syntax change?

**Comment by ccordoba12** [comment]
@krassowski, could you check what's happening with `jsonchema`?

**Comment by krassowski** [comment]
I opened https://github.com/Julian/jsonschema/issues/889. A workaround for now would be to pass an empty instance of settings schema (weather through standard input or from a file).

**Comment by ccordoba12** [comment]
> I opened Julian/jsonschema#889.

Thanks for your help @krassowski!

> A workaround for now would be to pass an empty instance of settings schema (weather through standard input or from a file).

What do you mean by this?

**Comment by krassowski** [comment]
I mean something like:

```bash
echo "{}" | jsonschema pylsp/config/schema.json
```

because an empty instance (empty settings) are acceptable, so they will pass validation.

Alternatively we could pin a 3.x version. Ether should solve the problem on the CI for now.

**Comment by npradeep357** [comment]
Updated jsonschema test as mentioned by @krassowski. Now the tests pass.
@krassowski or @ccordoba12 can you please review and approve this PR.

Thank you

File: README.md [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 [![image](https://github.com/python-ls/python-ls/workflows/Linux%20tests/badge.svg)](https://github.com/python-ls/python-ls/actions?query=workflow%3A%22Linux+tests%22) [![image](https://github.com/python-ls/python-ls/workflows/Mac%20tests/badge.svg)](https://github.com/python-ls/python-ls/actions?query=workflow%3A%22Mac+tests%22) [![image](https://github.com/python-ls/python-ls/workflows/Windows%20tests/badge.svg)](https://github.com/python-ls/python-ls/actions?query=workflow%3A%22Windows+tests%22) [![image](https://img.shields.io/github/license/python-ls/python-ls.svg)](https://github.com/python-ls/python-ls/blob/master/LICENSE)
 
-A Python 3.6+ implementation of the [Language Server Protocol](https://github.com/Microsoft/language-server-protocol).
+A Python 3.7+ implementation of the [Language Server Protocol](https://github.com/Microsoft/language-server-protocol).
+(Note: Python 3.6 is now not supported)
```
Comment (krassowski) [inline README.md]: Minor: I don't feel it's needed - just repeats the same information as in Python 3.7+. Above. I suggest to remove it or replace with something useful like:

```suggestion
(Note: versions <1.4 should still work with Python 3.6)
```

**Review by krassowski — APPROVED** [review]
Looks good to me. Thanks! I don't know if we want to merge immediately or wait until the EOL of Python 3.6 (23 Dec 2021).

**Comment by krassowski** [comment]
Merging as Python 3.6 reached EOL yesterday.

**Pull request merged**